### PR TITLE
WT-12086 Investigate replacing macros with inline functions

### PR DESCRIPTION
**Pull request by jeremythorp**

### CODE REVIEW

File: src/include/error.h [hunk 1]
```diff
@@ -98,22 +98,34 @@
...
-    } while (0)
-#define WT_TRET_ERROR_OK(a, e)                                                               \
-    do {                                                                                     \
-        int __ret;                                                                           \
-        if ((__ret = (a)) != 0 && __ret != (e) &&                                            \
-          (__ret == WT_PANIC || ret == 0 || ret == WT_DUPLICATE_KEY || ret == WT_NOTFOUND || \
-            ret == WT_RESTART))                                                              \
-            ret = __ret;                                                                     \
-    } while (0)
+__inline static void
```
[the remark this answers is not in the document]
Comment (jeremythorp) [reply src/include/error.h:101]: Good point! inline makes more sense.

[the remark this answers is not in the document]
Comment (jeremythorp) [reply src/include/error.h]: I've make that change.

File: src/rollback_to_stable/rts_btree.c [hunk 2]
```diff
@@ -321,9 +321,9 @@ __rts_btree_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
      * have moved this value to the history store as a full value. Therefore, we can safely ignore
      * the on page value if it is overflow removed.
      */
-    if (__wt_cell_type_raw(unpack->cell) == WT_CELL_VALUE_OVFL_RM)
+    if (__wt_cell_type_raw(unpack->cell) == WT_CELL_VALUE_OVFL_RM) {
```
Comment (ershov) [inline src/rollback_to_stable/rts_btree.c:324]: Why added braces?

File: test/unittest/tests/test_prepare_mod_sort.cpp [hunk 3]
```diff
@@ -118,9 +118,9 @@ init_key(WT_SESSION_IMPL *session, WT_ITEM *key, std::string key_str)
     WT_DECL_RET;
 
     ret = __wt_buf_init(session, key, key_str.size());
-    WT_ASSERT(session, ret == 0);
+    REQUIRE(ret == 0);
```
Comment (ershov) [inline test/unittest/tests/test_prepare_mod_sort.cpp:121]: Does it belong to this PR?

**Review by ershov — APPROVED** [review]
Clean work! Thanks Jeremy.
Just a few minor comments here.

File: test/unittest/tests/test_prepare_mod_sort.cpp (hunk 3 above)
Comment (jeremythorp) [reply test/unittest/tests/test_prepare_mod_sort.cpp:121]: A good question, and not necessarily. It was a mistake I picked up as I was looking through the code. I could move it into a seperate ticket/PR.

File: src/rollback_to_stable/rts_btree.c (hunk 2 above)
Comment (jeremythorp) [reply src/rollback_to_stable/rts_btree.c:324]: That was picked up by one of the code analysers.

Comment (jeremythorp) [reply src/rollback_to_stable/rts_btree.c:324]: I thought that came from a code analyser, but actually it was needed for a temporary change that I've now removed. I'll remove the braces.

Comment (jeremythorp) [reply src/rollback_to_stable/rts_btree.c]: Now removed.

File: test/unittest/tests/test_prepare_mod_sort.cpp (hunk 3 above)
Comment (jeremythorp) [reply test/unittest/tests/test_prepare_mod_sort.cpp]: I've rolled back this change, and will create a separate ticket for it.

Comment (jeremythorp) [reply test/unittest/tests/test_prepare_mod_sort.cpp]: I've created https://jira.mongodb.org/browse/WT-12364 to make this change.

**Pull request merged**